sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,742 @@
+    (runtime (nlogn n u4 u1780)))
+
+(define-read-only (cost_tuple_merge (n uint))
+    (runtime (linear n u208 u185)))
+
+(define-read-only (cost_tuple_cons (n uint))
+    (runtime (nlogn n u11 u1481)))
+
+(define-read-only (cost_add (n uint))
+    (runtime (linear n u11 u152)))
+
+(define-read-only (cost_sub (n uint))
+    (runtime (linear n u11 u152)))
+
+(define-read-only (cost_mul (n uint))
+    (runtime (linear n u12 u151)))
+
+(define-read-only (cost_div (n uint))
+    (runtime (linear n u13 u151)))
+
+(define-read-only (cost_geq (n uint))
+    (runtime u162))
+
+(define-read-only (cost_leq (n uint))
+    (runtime u164))
+
+(define-read-only (cost_le (n uint))
+    (runtime u152))
+
+(define-read-only (cost_ge (n uint))
+    (runtime u152))
+
+(define-read-only (cost_int_cast (n uint))
+    (runtime u157))
+
+(define-read-only (cost_mod (n uint))
+    (runtime u166))
+
+(define-read-only (cost_pow (n uint))
+    (runtime u166))
+
+(define-read-only (cost_sqrti (n uint))
+    (runtime u165))
+
+(define-read-only (cost_log2 (n uint))
+    (runtime u156))
+
+(define-read-only (cost_xor (n uint))
+    (runtime u163))
+
+(define-read-only (cost_not (n uint))
+    (runtime u158))
+
+(define-read-only (cost_eq (n uint))
+    (runtime (linear n u8 u155)))
+
+(define-read-only (cost_begin (n uint))
+    (runtime u189))
+
+(define-read-only (cost_secp256k1recover (n uint))
+    (runtime u14312))
+
+(define-read-only (cost_secp256k1verify (n uint))
+    (runtime u13488))
+
+(define-read-only (cost_some_cons (n uint))
+    (runtime u217))
+
+(define-read-only (cost_ok_cons (n uint))
+    (runtime u209))
+
+(define-read-only (cost_err_cons (n uint))
+    (runtime u205))
+
+(define-read-only (cost_default_to (n uint))
+    (runtime u255))
+
+(define-read-only (cost_unwrap_ret (n uint))
+    (runtime u330))
+
+(define-read-only (cost_unwrap_err_or_ret (n uint))
+    (runtime u319))
+
+(define-read-only (cost_is_okay (n uint))
+    (runtime u275))
+
+(define-read-only (cost_is_none (n uint))
+    (runtime u229))
+
+(define-read-only (cost_is_err (n uint))
+    (runtime u268))
+
+(define-read-only (cost_is_some (n uint))
+    (runtime u217))
+
+(define-read-only (cost_unwrap (n uint))
+    (runtime u281))
+
+(define-read-only (cost_unwrap_err (n uint))
+    (runtime u273))
+
+(define-read-only (cost_try_ret (n uint))
+    (runtime u275))
+
+(define-read-only (cost_match (n uint))
+    (runtime u316))
+
+(define-read-only (cost_or (n uint))
+    (runtime (linear n u3 u147)))
+
+(define-read-only (cost_and (n uint))
+    (runtime (linear n u3 u146)))
+
+(define-read-only (cost_append (n uint))
+    (runtime (linear n u1 u1024)))
+
+(define-read-only (cost_concat (n uint))
+    (runtime (linear n u1 u1004)))
+
+(define-read-only (cost_as_max_len (n uint))
+    (runtime u482))
+
+(define-read-only (cost_contract_call (n uint))
+    (runtime u154))
+
+(define-read-only (cost_contract_of (n uint))
+    (runtime u13391))
```

Comment:
There is an open issue on the benchmark repo suggesting to set this value to less than 1000:
https://github.com/blockstack/clarity-benchmarking/issues/22

Reply:
Yes -- I think both of those issues could change the final proposal here. They'll need to be addressed first via the clarity-benchmarking repository (and it's output) before updating here.